packages/elements-core/src/context/Options.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+
+import type { DocsProps } from '../components/Docs';
+
+const DEFAULT_CONTEXT: ElementsOptionsContextProps = {};
+
+export type ElementsOptionsContextProps = Pick<DocsProps, 'nodeHasChanged'>;
+
+export const ElementsOptionsContext = React.createContext<ElementsOptionsContextProps | undefined>(undefined);
+
+export const useOptionsCtx = () => {
+  return React.useContext(ElementsOptionsContext) || DEFAULT_CONTEXT;
+};
+
+export type ProviderProps = Partial<ElementsOptionsContextProps> & {
+  children: React.ReactNode;
+};
+
+export function ElementsOptionsProvider({ children, nodeHasChanged }: ProviderProps) {
```

Comment:
Decided to add an options provider to avoid having to pass top level options through the entire component tree everywhere. Just using it for `nodeHasChanged` via `const { nodeHasChanged } = useOptionsCtx();` at the moment, but we can easily use this for other options in the future to cut down on the manual prop passing.